index.js

```diff
@@ -1,37 +1,36 @@
 'use strict';
 
-var fnToStr = Function.prototype.toString;
-
 var constructorRegex = /^\s*class\b/;
-var isES6ClassFn = function isES6ClassFunction(value) {
-	try {
-		var fnStr = fnToStr.call(value);
-		return constructorRegex.test(fnStr);
-	} catch (e) {
-		return false; // not a function
-	}
-};
+var fnToStr = Function.prototype.toString;
+var objToStr = Object.prototype.toString;
+var fnClassStr = '[object Function]';
+var genClassStr = '[object GeneratorFunction]';
+var hasToStringTag = typeof Symbol === 'function' && typeof Symbol.toStringTag === 'symbol';
 
-var tryFunctionObject = function tryFunctionToStr(value) {
+var getFnStr = function tryFnStr(value) {
 	try {
-		if (isES6ClassFn(value)) { return false; }
-		fnToStr.call(value);
-		return true;
+		return fnToStr.call(value);
 	} catch (e) {
-		return false;
+		return null;
 	}
 };
-var toStr = Object.prototype.toString;
-var fnClass = '[object Function]';
-var genClass = '[object GeneratorFunction]';
-var hasToStringTag = typeof Symbol === 'function' && typeof Symbol.toStringTag === 'symbol';
 
 module.exports = function isCallable(value) {
 	if (!value) { return false; }
-	if (typeof value !== 'function' && typeof value !== 'object') { return false; }
-	if (typeof value === 'function' && !value.prototype) { return true; }
-	if (hasToStringTag) { return tryFunctionObject(value); }
-	if (isES6ClassFn(value)) { return false; }
-	var strClass = toStr.call(value);
-	return strClass === fnClass || strClass === genClass;
+
+	var tmp = typeof value;
+	if (tmp === 'function') {
```

Comment:
storing the typeof value actually defeats engine optimizations in many engines (at least some older ones). it's better to repeat the `typeof value ===` every time.

Reply:
Huh, didnt know.